std-out only

braver opened the pull request:
This PR primarily removes the GitHub token. We can't manage secrets on https://github.com/wbond/package_control_channel, so if we're to move away from whatever ancient script resides on the packagecontrol.io website, this is the only way forward (without moving away from the channel repo completely). Besides, the bot was extremely unreliable anyway, sometimes it doesn't run on a PR at all, other times it fails without feedback. 

This means with this version output goes to std-out and throws an error if the package isn't compliant.

The downside is you don't get a nice message in the PR. Edit: you do get annotations with clear messages.
The upside though is that you can use the script locally in a terminal easily and it works just the same way (ie. the GitHub interaction doesn't complicate the tool and there is no additional testing required).
... anyway, don't think we have a choice. 

This also:
- updates some tests because the existing version is deeply outdated.
- updates the vendored version of the actual package reviewing script to its latest version.
- we cannot upgrade the vendored package control package code because that was never intended to be used this way and its "API" if you will has completely changed. 
- update keymaps to recent version of ST

From some testing it looks like this is as good as the existing bot/job on the pc channel for simple package additions. And it's just as bad at resolving more complex ones (e.g. https://github.com/wbond/package_control_channel/pull/9178). So that's good: status quo is maintained functionally, but we now own the code and can improve it. 

This PR isn't intended as something to merge. As far as I understand only the [LSP](https://github.com/sublimelsp/repository) project uses this, and perhaps one or two individuals. Package authors probably mostly use the standalone [package reviewer](https://github.com/packagecontrol/st_package_reviewer) if they use anything at all. In any case I don't mean to upset what everyone else is doing. Perhaps after we've improved things or the entire status quo is resolved.

braver committed:
remove sublime-syntax check

we recommend to ignore it anyway, 3092 is deeply ancient

braver committed:
remove PR comments

we don't have a token to work with anyway,
lets focus on good stdout/stderr output

braver committed:
no longer need a token to run

braver committed:
remove make vague recommendations around licenses

braver committed:
remove ast based checkers

these are both complex and hard to maintain,
and of relatively limited use

braver committed:
downloader settings are effectively a constant

also avoids the yml dependency

braver committed:
remove now unused args

braver committed:
the more you know...

braver committed:
repo url no longer used

braver committed:
out to stdout/err

braver committed:
update st_package_reviewer lib

braver committed:
specifying all platforms is a reason to reject

braver committed:
readme is not optional

braver committed:
remove release tagging job

braver committed:
silence debug output for downloader

braver committed:
update readme

braver committed:
update outdated keymaps

braver committed:
update known keys list

braver committed:
ignore sublime project/workspace

braver committed:
missed "find"

michaelblyons commented on lib/st_package_reviewer/check/file/check_messages.py:
```diff
@@ -38,11 +27,10 @@ def check(self):
                 if key == "install":
                     pass
-                elif SemVer.valid(re.sub(prefix_regex, '', key)):
+                elif SemVer.valid(key):
```
Is this why you thought my USFM package was bad?

braver replied:
No. I did not make that change. This is the [st_package_reviewer](https://github.com/packagecontrol/st_package_reviewer) that's vendored here. And this is just an excellent example of why vendoring is bad 😉 

https://github.com/packagecontrol/st_package_reviewer/blame/master/st_package_reviewer/check/file/check_messages.py#L33

That's 9 years old.

braver replied:
https://github.com/packagecontrol/st-package-reviewer-action/commit/cd814a8c809814b2fcbbf6733a85149e962a9902#diff-73d93f3cc739df727f2bf784becacddc3b5622e61b1d6c3de6790ebcd97e2aa9R45 has the original. 

Perhaps that wasn't just a straight copy but modified somehow secretly?

braver replied:
Ugh... semi-related 657279367ef7f9e216b9360d7b1b6c8e40964f65 should also not really be committed here but upstream. I wish this was just a proper git submodule or something sensible.

michaelblyons replied:
> No. I did not make that change.

Sorry. Didn't mean to accuse. :sweat_smile:

braver replied:
No problem! It's good you caught this. I'm a little unsure what to do about this... perhaps we need some PR's upstream and then re-align.

braver replied:
I brought the prefix handling back. Perhaps I'll just try to make something that works and we can worry about what to do about the package reviewer dependency later. This needs re-architecting anyway, and that's not what this PR is about.

michaelblyons reviewed (changes requested):
You broke the semver prefix check.

Also, not a blocker, but a nice-to-have: GitHub [has a format](https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message) for identifying errors and warnings and including a traceback in the Check output.

braver committed:
revert check_messages in the package reviewer dep

braver committed:
immediately print output in github's format

https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-commands#setting-a-notice-message

braver committed:
simply immediately print everything

braver committed:
bring back checker report formatting

braver commented:
Partial output (due to GH API rate limiting) for @FichteFoll's recent PR to move stuff out of his personal repo:

```sh
% python3 ../st-package-reviewer-action/run-review-bot.py --base-sha e88cec07a537a070ce504fa7a553ef0111d46b8c --current-sha b4a60e96e937f8dd06a3d5dc594932a92106a075
::notice title=REPO_ADDED ::https://raw.githubusercontent.com/FichteFoll/sublime_packages/master/package_control.json
::notice title=ADDED ::InsertDate, InactivePanes, File History, CSScheme, QuickRulers
::warning title=CHECK ::Command class 'create_csscheme' does not end with 'Command': File: create_new_csscheme.py, Line: 156, Column: 1
::warning title=CHECK ::The command 'create_csscheme' is not PascalCase: File: create_new_csscheme.py, Line: 156, Column: 1
::warning title=CHECK ::Command class 'SlApplyEdit' does not end with 'Command': File: my_sublime_lib/edit.py, Line: 166, Column: 5
::warning title=CHECK ::Found multiple command prefixes: Sl, Window. Consider using one single prefix so as to not clutter the command namespace.
::notice title=PASS ::CSScheme
::warning title=CHECK ::Found multiple command prefixes: Cleanup, Open, Reset. Consider using one single prefix so as to not clutter the command namespace.
::error title=CHECK ::Key '' is not 'install' or a valid semantic version: File: messages.json
::error title=CHECK ::File '', as specified by key '', does not exist: File: messages.json
::error title=CHECK ::The binding ['ctrl+shift+t'] unconditionally overrides a default binding: File: Default (Windows).sublime-keymap
::error title=CHECK ::The binding ['super+shift+t'] unconditionally overrides a default binding: File: Default (OSX).sublime-keymap
::error title=CHECK ::The binding ['ctrl+shift+t'] unconditionally overrides a default binding: File: Default (Linux).sublime-keymap
::warning title=CHECK ::The binding ['right'] is also defined in default bindings but is masked with a 'context': File: Default (Windows).sublime-keymap
::warning title=CHECK ::The binding ['ctrl+delete'] is also defined in default bindings but is masked with a 'context': File: Default (Windows).sublime-keymap
::warning title=CHECK ::The binding ['super+delete'] is also defined in default bindings but is masked with a 'context': File: Default (OSX).sublime-keymap
::warning title=CHECK ::The binding ['right'] is also defined in default bindings but is masked with a 'context': File: Default (Linux).sublime-keymap
::warning title=CHECK ::The binding ['ctrl+delete'] is also defined in default bindings but is masked with a 'context': File: Default (Linux).sublime-keymap
Package Control: Hit rate limit of 60 for api.github.com. Skipping all futher download requests for this domain.
::error title=FAIL ::Rate limit of 60 exceeded for api.github.com
::error title=FAIL ::Skipping due to hitting rate limit for api.github.com
::error title=FAIL ::Skipping due to hitting rate limit for api.github.com
Errors occurred
```

braver committed:
fix added/removed were flipped

braver committed:
titles are not always displayed

braver committed:
ok

FichteFoll commented:
I do not have the capacity to review this in full currently, but the direction of using GitHub Actions and stdout printing is definitely the correct one (and we have discussed that in the past). I'm not familiar with the old state of this code base, nor can I compare it against what st_package_reviewer currently looks like off the top of my head, but from a quick glance there does not appear to be anything problematic in these changes.

Since the current repo is mostly used by LSP, as far as I am aware, they (or @rchl in particular) should be the ones to review and merge this. From a main channel perspective, I am fine with pointing the action there to a particular commit on this potentially longer-running branch. Since you will be the primary consumer/user of these changes, I trust you to have done the changes with good usability in mind, but feel free to tag me for PRs against this branch in the future and I will see if I find the time to take a look.

rchl commented on run-review-bot.py:
```diff
@@ -1,17 +1,19 @@
-    res = test_pull_request(args.pr_url, args.base_sha, args.current_sha, args.token)
-    if res['result'] == 'error':
-        raise Exception(res['message'])
+    res = test_pull_request(args.base_sha, args.current_sha)
```
For readability would do:

```suggestion
    exit_code, message = test_pull_request(args.base_sha, args.current_sha)
```

rchl commented on lib/run_repo_tests.py:
```diff
@@ -369,187 +324,88 @@ def test_pull_request(pr_url: str, old_rev: str, current_rev: str, token: str):
+                    print('::notice title=PASS ::OK {}'.format(name))
+                else:
+                    success = False
+                continue
```
`continue` can be removed

rchl commented on lib/run_repo_tests.py:
```diff
@@ -57,59 +46,65 @@ def run_tests(spec):
-
         if 'sublime' in info['name'].lower():
-            errors.append('Package name contains the word "sublime"')
+            print('::error title=NAME ::Package name contains the word Sublime')
```
`...word "sublime"` phrasing was better/more correct/specific. I would keep the quotes basically.

rchl commented on lib/st_package_reviewer/runner.py:
```diff
@@ -0,0 +1,64 @@
+    def result(self):
+        """Return whether checks ran without issues (`True`) or there were failures (`False`)."""
+        if not self._checked:
+            raise RuntimeError("Check has not been perfomed yet")
```
```suggestion
            raise RuntimeError("Check has not been performed yet")
```

rchl commented on README.md:
```diff
@@ -1,57 +1,33 @@
         with:
-          python-version: 3.6
-
-      - name: Install Python dependencies
-        run: python3 -m pip install pyyaml
```
Wouldn't we still need to install some dependencies. Like `github3.py` for example?

rchl replied:
BTW. Why specific choice of `github3.py`? One downside of it that https://github.com/PyGithub/PyGithub doesn't have for example is that it doesn't include types. While this code doesn't use types, I think it would be natural next step to add them for improved robustness.

braver replied:
That kind of feedback should be aimed at https://github.com/packagecontrol/st_package_reviewer. This action just vendors and wraps the scripting in that repo. 

> Wouldn't we still need to install some dependencies

I guess not, since we're actually using the PC package code to get us a zipped package, so _that_ interacts with GitHub instead of st_package_reviewer (which could also do it). 

Basically, architecturally this thing is a mess, but it works.

rchl commented on lib/st_package_reviewer/__main__.py:
```diff
@@ -0,0 +1,236 @@
+
+from github3 import GitHub
+
+from . import set_debug, debug_active, __version__
```
Importing some non-existent stuff. Seems unfinished/untested.

braver replied:
Note that you might be reviewing the st_package_reviewer dependency instead of the action that wraps it. And that dependency is currently a mishmash of versions of the package reviewer script...

rchl replied:
OK but you've added that file and it contains imports that don't exist - `set_debug`

braver replied:
I just copied it over from https://github.com/packagecontrol/st_package_reviewer/blob/master/st_package_reviewer/__main__.py in an attempt to sync up with the source of that script. But sadly the st_package_reviewer dir is not actually any version of https://github.com/packagecontrol/st_package_reviewer, it's deviated. This is a mess I intend to figure out and clean up upstream, but for now it does not seem to affection functionality. In other words, the only thing that should matter there, in this repo and this PR is `lib/run_repo_tests.py`, everything else is an unresolved cludge. Did I mention it's not intended to be merged? 😉 

> unfinished/untested

It's definitely unfinished, but it cannot be finished without resolving the deviations from [st_package_reviewer](https://github.com/packagecontrol/st_package_reviewer). It is however quite tested, on the command line and in practice over at the default channel. 

Much of what looks "off" is not actually a problem in practice, in this GitHub action use case, because we're not using the full interface of st_package_reviewer: we exclusively feed it resolved packages.

rchl reviewed (commented):
> This PR isn't intended as something to merge. 

Why not? I suppose we'd be fine with this way of presenting results in LSP also. (As long as normal users can also access those results).

braver committed:
typo

Co-authored-by: Rafał Chłodnicki <rchl2k@gmail.com>
